### PR TITLE
implement auto-sort for UniqueListingList

### PR DESCRIPTION
**Pull request by hoang227**

### CODE REVIEW

File: src/main/java/seedu/address/model/listing/Listing.java [hunk 1]
```diff
@@ -160,6 +160,14 @@ public boolean isSameListing(Listing otherListing) {
                 || otherListing.id.equals(this.id));
     }
 
+    /**
+     * Compare current listing to l
```
Comment (jeromehjj) [inline src/main/java/seedu/address/model/listing/Listing.java:164]:
```suggestion
     * Compares current listing to l
```

File: src/main/java/seedu/address/model/listing/UniqueListingList.java [hunk 2]
```diff
@@ -46,7 +46,11 @@ public void add(Listing toAdd) {
         if (contains(toAdd)) {
             throw new DuplicateListingException();
         }
-        internalList.add(toAdd);
+        int idx = 0;
```
Comment (jeromehjj) [inline src/main/java/seedu/address/model/listing/UniqueListingList.java:49]: I think maybe can change to `index` to make it clearer

**Review by jeromehjj — CHANGES_REQUESTED** [review]
Just some minor changes and then LGTM!

File: src/main/java/seedu/address/model/listing/Listing.java [hunk 3]
```diff
@@ -160,6 +160,14 @@ public boolean isSameListing(Listing otherListing) {
                 || otherListing.id.equals(this.id));
     }
 
+    /**
```
Comment (Gavzzz) [inline src/main/java/seedu/address/model/listing/Listing.java:163]: In line 29 isit possible to change  private final List<Person> interestedClients, to List<Client> interestedClients to be more specific?

**Review by Gavzzz — APPROVED** [review]

**Review by jeromehjj — APPROVED** [review]
LGTM!

**Pull request merged**